router/pkg/metric/metric_store.go

```diff
@@ -163,7 +167,7 @@ type (
 
 // NewStore creates a new metrics store instance.
 // The store abstract OTEL and Prometheus metrics with a single interface.
-func NewStore(opts ...Option) (Store, error) {
+func NewStore(otlpOpts MetricOpts, promOpts MetricOpts, opts ...Option) (Store, error) {
```

Comment:
This is the first time we actually enable/disable metrics within the same store, previously it was at the store level.